src/uwtools/config/tools.py

```diff
@@ -173,32 +171,30 @@ def _print_config_section(config: dict, key_path: List[str]) -> None:
 
 def _realize_config_update(
     config_obj: Config,
-    values: Union[dict, Config, OptionalPath],
-    values_format: Optional[str] = None,
+    supplemental_configs: Optional[List[Union[dict, Config, DefinitePath]]] = None,
```

Comment:
Since we repeat this complex type several times, and since all need to match, we could factor this out as a module-level type, e.g.
```
SupplementalConfigs = Optional[List[Union[dict, Config, DefinitePath]]]
```
near the top of the module, then use `SupplementalConfigs` in place of the complex type. If we need the same across modules, it could be factored out to `uwtools.types` (where `DefinitePath` and `OptionalPath` are defined). I have mixed feelings about this because it does mean having to go look up what the type means when you've forgotten, but this one seems tricky enough to potentially justify that. Your call.

Reply:
Since `SupplementalConfigs` requires `Config` to be imported, it creates a circular dependency. I'm going to leave it as-is for now, but I, too, have mixed feeling about the complex type.